app/actions/websocket/preferences.ts

```diff
@@ -76,3 +91,22 @@ async function handleSavePostAdded(serverUrl: string, preferences: PreferenceTyp
         }
     }
 }
+
+const differsFromLocalNameFormat = async (serverUrl: string, preferences: PreferenceType[]) => {
+    const database = DatabaseManager.serverDatabases[serverUrl]?.database;
+    if (!database) {
+        return false;
+    }
+
+    const displayPref = getPreferenceValue(preferences, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.NAME_NAME_FORMAT) as string;
+    if (displayPref === '') {
+        return false;
+    }
+
+    const currentPref = await queryPreferencesByCategoryAndName(database, Preferences.CATEGORY_DISPLAY_SETTINGS, Preferences.NAME_NAME_FORMAT, displayPref).fetch();
+    if (currentPref.length > 0) {
+        return false;
+    }
+
+    return true;
+};
```

Comment:
consider moving this to the queries folder and get the database as the first parameter instead of the serverUrl, I don't think this function can be considered an action as is not changing anything or fetching anything, and even if it would be an action, I don't think it belongs to this file either.